etna/transforms/feature_selection/feature_importance.py

```diff
@@ -229,13 +229,17 @@ def _fit(self, df: pd.DataFrame) -> "MRMRFeatureSelectionTransform":
             instance after fitting
         """
         features = self._get_features_to_use(df)
-        ts = TSDataset(df=df, freq=pd.infer_freq(df.index))
-        relevance_table = self.relevance_table(ts[:, :, "target"], ts[:, :, features], **self.relevance_params)
+        df_target = df.loc[:, pd.IndexSlice[:, "target"]]
+        df_target = df_target.loc[df_target.first_valid_index() :]
+        df_features = df.loc[:, pd.IndexSlice[:, features]]
+        df_features = df_features.loc[df_features.first_valid_index() :]
```

Comment:
Why we need `first_valid_index` here?

Reply:
Because previously we used `ts[:, :, :]` indexing which makes `first_valid_index` logic, you could check it [here](https://github.com/etna-team/etna/blob/master/etna/datasets/tsdataset.py#L229).